test/multi_node_test_2.py

```diff
@@ -34,6 +43,30 @@ def test_cluster():
     assert not any(node.any_processes_alive() for node in g.list_all_nodes())
 
 
+def test_internal_config():
+    cluster = Cluster(
+        initialize_head=True,
+        connect=True,
+        head_node_args={
+            "resources": dict(CPU=1),
+            "_internal_config": json.dumps({
+                "num_heartbeats_timeout": 20
+            })
+        })
+    worker = cluster.add_node()
+    cluster.wait_for_nodes()
+
+    cluster.remove_node(worker)
+    cluster.wait_for_nodes(retries=10)
+    assert ray.global_state.cluster_resources()["CPU"] == 2
+
+    cluster.wait_for_nodes(retries=20)
+    assert ray.global_state.cluster_resources()["CPU"] == 1
+    # The code after the yield will run as teardown code.
+    ray.shutdown()
+    cluster.shutdown()
```

Comment:
`ray.shutdown/cluster.shutdown` should be in a pytest fixture so that it happens even if the test fails.